.github/workflows/go-tests.yml

```diff
@@ -408,17 +408,19 @@ jobs:
       consul-license: ${{secrets.CONSUL_LICENSE}}
       datadog-api-key: "${{ !endsWith(github.repository, '-enterprise') && secrets.DATADOG_API_KEY || '' }}"
 
-  go-test-api-1-19:
+  go-test-api-backwards-compatibility:
+    name: go-test-api-${{ needs.get-go-version.outputs.go-version-previous }}
```

Comment:
Prior art: https://github.com/hashicorp/consul/pull/10488/files

The second commit in this PR (https://github.com/hashicorp/consul/pull/20062/commits/e71ae3aed88be120439ab3cdfa005e2b48ec1abf) automates the version selection so we don't have to remember to bump it every time we upgrade the minor version of Go.